openmmtools/mcmc.py

```diff
@@ -469,19 +465,19 @@ def apply(self, thermodynamic_state, sampler_state):
         move.apply(thermodynamic_state, sampler_state)
 
     def __getstate__(self):
-        serialization = dict()
-        for i, (move, weight) in enumerate(self.move_set):
-            serialization['move' + str(i)] = utils.serialize(move)
-            serialization['move' + str(i) + '_weight'] = weight
-        return serialization
+        serialized_moves = [utils.serialize(move) for move, _ in self.move_set]
+        weights = [weight for _, weight in self.move_set]
+        return dict(moves=serialized_moves, weights=weights)
```

Comment:
They're not meant to be passed to each other, so it should be fine. Every object only cares about its own representation.